api.go

```diff
@@ -207,23 +207,14 @@ func (api *API) addResource(prototype jsonapi.MarshalIdentifier, source interfac
 	}
 
 	var ptrPrototype interface{}
-	var name string
 
 	if resourceType.Kind() == reflect.Struct {
 		ptrPrototype = reflect.New(resourceType).Interface()
-		name = resourceType.Name()
 	} else {
 		ptrPrototype = reflect.ValueOf(prototype).Interface()
-		name = resourceType.Elem().Name()
 	}
 
-	// check if EntityNamer interface is implemented and use that as name
-	entityName, ok := prototype.(jsonapi.EntityNamer)
```

Comment:
this also should be saved, it looks similar that is done in function `getStructType `

Reply:
restored this behaviour